tests/models/deta/test_modeling_deta.py

```diff
@@ -520,6 +520,10 @@ def test_initialization(self):
                         msg=f"Parameter {name} of model {model_class} seems not properly initialized",
                     )
 
+    @unittest.skip("Cannot be initialized on meta device as some weights are modified during the initialization")
```

Comment:
This came from an existing comment:
https://github.com/huggingface/transformers/blob/354775bc5755c4a6c47e008d28f27f8ccdcf8f8f/src/transformers/models/deta/modeling_deta.py#L1780

It seems like the DetaForObjectDetection model has logic to manipulate the params during init, and so it simply doesn't support initializing on a meta device.

With that said, should we add some kind of flag in PreTrainedModel that would throw an error when trying to initial on meta device for models that do not support it? So far, it seems like DetaModel is the only one, so maybe not worth the effort to do so. 

cc @SunMarc who added this: https://github.com/huggingface/transformers/pull/27089

Reply:
Let's keep as-is for the moment, until we have more affected models. DETR and DETR-like models are special but I think they're the only case.